conans/assets/templates/new_v2_bazel.py

```diff
@@ -24,28 +26,30 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def validate(self):
-        if self.settings.os in ("Windows", "Macos") and self.options.shared:
-            raise ConanException("Windows and Macos needs extra BUILD configuration to be able "
-                                 "to create a shared library. Please, check this reference to "
-                                 "know more about it: https://bazel.build/reference/be/c-cpp")
-
     def layout(self):
         bazel_layout(self)
         # DEPRECATED: Default generators folder will be "conan" in Conan 2.x
         self.folders.generators = "conan"
 
     def build(self):
         bazel = Bazel(self)
-        bazel.build()
+        # Linux creates both shared and static libraries by default
+        if self.settings.os == "Linux":
+            bazel.build()
+        else:
+            if self.options.shared:
+                # main/BUILD uses cc_shared_library to create shared libraries
+                bazel.build(args=["--experimental_cc_shared_library"])
+            else:
+                bazel.build(target="//main:{name}")
```

Comment:
Is this ``{name}`` intended? What does it mean?

Reply:
This `{name}` is part of the template: `template.format(name="mypkg")`. That replacement is made below.